### PR TITLE
added cpp solution for usaco-623

### PR DESCRIPTION
**Pull request by RedBlueBird**
_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which include the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-623.mdx [hunk 1]
```diff
@@ -0,0 +1,115 @@
+---
+id: usaco-623
+source: USACO Gold 2016 February
+title: Fenced In
+author: Jeffrey Zhang
+---
+
+[Official Analysis](http://www.usaco.org/current/data/sol_fencedin_gold_feb16.html)
```
Comment (SansPapyrus683) [inline solutions/usaco-623.mdx:8]:
```suggestion
[Official Analysis (C++)](http://www.usaco.org/current/data/sol_fencedin_gold_feb16.html)
```

File: solutions/usaco-623.mdx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+## Solution
+
+The $n$ vertical and $m$ horizontal fences divide the large rectangular pasture into $(n+1)(m+1)$ grids. To help us better comprehend the problem, it is easier for us to visualized the input in terms of graphs: the grids themselves are the nodes, and the fence lengths of each 4 surrounding sides are the edge lengths that can be constructed to their neighboring nodes. For a cow to travel from any node to another, all we need is to construct a tree graph. And to ensure the edges picked are at a minimum, we just need to construct a minimum spanning tree, which can be solved by the [Kruskal's MST algorithm](https://usaco.guide/gold/mst#kruskals).
```
Comment (SansPapyrus683) [inline solutions/usaco-623.mdx:12]: this seems to just repeat the official?

File: solutions/usaco-623.mdx [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    }
+
+    // Perform the MST
+    ll ans = 0;
+    for (int x = 0, y = 0; x <= n || y <= m; ) {
+        if (x <= n && vert[x] < horz[y] || y > m) {
+            for (int i = 0; i < m; i++) {
+                if (merge(make_pair(x, i), make_pair(x, i + 1), top, size)) {
+                    ans += vert[x];
+                }
```
Comment (SansPapyrus683) [inline solutions/usaco-623.mdx:96]: tabs

File: solutions/usaco-623.mdx [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+        swap(a, b);
+    }
+    top[b.ff][b.ss] = a;
+    size[a.ff][a.ss] += size[b.ff][b.ss];
+    return true;
+}
+//EndCodeSnip
+
+int main() {
+    ofstream fout("fencedin.out");
```
Comment (SansPapyrus683) [inline solutions/usaco-623.mdx:53]: put this at the bottom where you actually use it

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-623.mdx (hunk 2 above)
Comment (RedBlueBird) [reply solutions/usaco-623.mdx:12]: The problem is in the MST module, and I don't think there are other ways to solve this...

Comment (SansPapyrus683) [reply solutions/usaco-623.mdx:12]: if that's the case, then just add an implementation- no need for a solution

**Pull request closed without merging**